cpp/src/groupby/sort/group_tdigest.cu

```diff
@@ -408,13 +422,100 @@ generate_group_cluster_info(int delta,
     cumulative_weight,
     group_cluster_wl.begin(),
     group_num_clusters.begin(),
-    group_cluster_offsets->view().begin<offset_type>());
+    group_cluster_offsets->view().begin<offset_type>(),
+    has_nulls);
 
   return {std::move(group_cluster_wl),
           std::move(group_cluster_offsets),
           static_cast<size_type>(total_clusters)};
 }
 
+std::unique_ptr<column> build_output_column(size_type num_rows,
+                                            std::unique_ptr<column>&& means,
+                                            std::unique_ptr<column>&& weights,
+                                            std::unique_ptr<column>&& offsets,
+                                            std::unique_ptr<column>&& min_col,
+                                            std::unique_ptr<column>&& max_col,
+                                            bool has_nulls,
+                                            rmm::cuda_stream_view stream,
+                                            rmm::mr::device_memory_resource* mr)
+{
+  // whether or not this weight is a stub
+  auto is_stub_weight = [weights = weights->view().begin<double>()] __device__(size_type i) {
+    return weights[i] == 0;
+  };
+  // whether or not this particular tdigest is a stub
+  auto is_stub_digest = [offsets = offsets->view().begin<offset_type>(), is_stub_weight] __device__(
+                          size_type i) { return is_stub_weight(offsets[i]) ? 1 : 0; };
```

Comment:
Silly nitpick: If this function is expected to return a bool value, should we simply do the following?
```suggestion
                          size_type i) { return is_stub_weight(offsets[i]); };
```

Reply:
>  If this function is expected to return a bool value

Nope! My mistake. We're aggregating these as `int32_t`. Please ignore.